common/lib/dependabot/group_rule.rb

```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+module Dependabot
+  class GroupRule
```

Comment:
I'm a little unclear on the interfact for a `GroupRule`. The `rule` portion signifies a policy-like approach, where criteria must be met. I'm open to this changing, or to refining the concept as we move forward.